src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -343,7 +344,11 @@ private void retrievePullRequests(SCMSourceCriteria criteria, SCMHeadObserver ob
                 checkInterrupt();
                 PullRequestSCMHead head = new PullRequestSCMHead(pull.getSource().getRepository().getOwnerName(),
                         pull.getSource().getRepository().getRepositoryName(), repositoryType,
-                        pull.getSource().getBranch().getName(), pull);
+                        pull.getSource().getBranch().getName(), pull,
+                        getRepoOwner().equalsIgnoreCase(pull.getSource().getRepository().getOwnerName())
```

Comment:
I think BB allows to fork repositories under the same user (`ownerName`) but using a different repo name. It would be recognized as `SCMHeadOrigin.DEFAULT` here, which I think is wrong.